src/server/src/main/java/io/cassandrareaper/service/RepairRunner.java

```diff
@@ -344,77 +282,48 @@ private void endRepairRun() {
    */
   private void startNextSegment() throws ReaperException, InterruptedException {
     boolean scheduleRetry = true;
-    boolean anythingRunningStill = false;
 
     // We want to know whether a repair was started,
     // so that a rescheduling of this runner will happen.
     boolean repairStarted = false;
 
-    for (int rangeIndex = 0; rangeIndex < currentlyRunningSegments.length(); rangeIndex++) {
-
-      if (currentlyRunningSegments.get(rangeIndex) != null) {
-        anythingRunningStill = true;
-
-        // Just checking that no currently running segment runner is stuck.
-        RepairSegment supposedlyRunningSegment
-            = context.storage.getRepairSegment(repairRunId, currentlyRunningSegments.get(rangeIndex)).get();
-        DateTime startTime = supposedlyRunningSegment.getStartTime();
-        if (startTime != null && startTime.isBefore(DateTime.now().minusDays(1))) {
-          LOG.warn(
-              "Looks like segment #{} has been running more than a day. Start time: {}",
-              supposedlyRunningSegment.getId(),
-              supposedlyRunningSegment.getStartTime());
-        } else if (startTime != null && startTime.isBefore(DateTime.now().minusHours(1))) {
-          LOG.info(
-              "Looks like segment #{} has been running more than an hour. Start time: {}",
-              supposedlyRunningSegment.getId(),
-              supposedlyRunningSegment.getStartTime());
-        } else if (startTime != null && startTime.isBefore(DateTime.now().minusMinutes(2))) {
-          LOG.debug(
-              "Looks like segment #{} has been running more than two minutes. Start time: {}",
-              supposedlyRunningSegment.getId(),
-              supposedlyRunningSegment.getStartTime());
-        }
-        // No need to try starting new repair for already active slot.
-        continue;
+    // We have an empty slot, so let's start new segment runner if possible.
+    // When in sidecar mode, filter on ranges that the local node is a replica for only.
+    LOG.info("Attempting to run new segment for range");
```

Comment:
can you print what the range is? please.

Reply:
This message is not relevant anymore and the range part needs to be removed. It was a reference to the parallelRanges that have been removed by this PR (we don't know yet which range we'll be processing next as the segment needs to be selected first).
I'll adjust the wording.